dlib/global_optimization/find_max_global_abstract.h

```diff
@@ -134,6 +146,7 @@ namespace dlib
             - find_max_global() runs until one of the following is true:
                 - The total number of calls to the provided functions is == num.max_calls
                 - More than max_runtime time has elapsed since the start of this function.
+                - should_stop(f(x)) returns true, the default never returns true
```

Comment:
Only say `should_stop(f(x)) returns true`.  The identity of the default should be communicated to the reader above in the signature.